### PR TITLE
Luacheck warnings inlining

### PR DESCRIPTION
**Pull request by tobylane**
**Describe what the proposed change does**
- Convert most of the luacheck file wide ignores to line specific
- Fix the other warnings

### CODE REVIEW

File: CorsixTH/Lua/dialogs/menu.lua [hunk 1]
```diff
@@ -753,7 +753,7 @@ function UIMenuBar:makeGameMenu(app)
   end
   local function overlay(...)
     local args = {n = select('#', ...), ...}
-    return function(item, m)
+    return function(item, m) -- luacheck: ignore 212
```
Comment (lewri) [inline CorsixTH/Lua/dialogs/menu.lua:756]: It's not needed, but was this one meant to have the full inline context as the other 212 ignores have?

**Review by lewri — COMMENTED** [review]
I'm guessing the functions where you replaced the first argument with `_` will be for compatibility reasons? 

Otherwise my read through the changes seems ok enough

**Comment by Alberth289346** [comment]
I am not sure why parameter information gets destroyed just because the local derived method doesn't use the value.

For plain functions it makes sense, since there is exactly one implementation.

In a class system with derived classes there are several implementations of the function. Keeping the function signature consistent across the classes helps in identifying the functions imho.

File: CorsixTH/Lua/dialogs/menu.lua (hunk 1 above)
Comment (tobylane) [reply CorsixTH/Lua/dialogs/menu.lua:756]: That function could be ..., I made this change as I understand it. It's local, no context.

**Comment by tobylane** [comment]
Yes, Luacheck doesn't know of classes. I've not taken parameters out of any class functions, aiming for that consistency.

**Comment by Alberth289346** [comment]
I don't exactly know about all functions, but my guess is that at least the interrupt functions and eg the doKey functions are all stored in the same spot and called in the same manner too.
So while technically not a method of a class, effectively they do have a fixed call signature. (That is, we might add them as methods rather than with ad-hoc callback functions if we would write that code today.)

The original author of a lot of the code preferred plain functions above class methods, or maybe at the time of writing there were no classes (I wasn't there at the time so I don't know).

**Comment by TheCycoONE** [comment]
Why don't we suppress that check globally. It appears more noisy than helpful on our code.

**Comment by tobylane** [comment]
Should the sprite viewer be treated as if it is a class?

Added the ignore. Removed the std because it was set to the default at the time, the default is now 5.1-4 plus JIT.

**Pull request merged**